test/test_other.py

```diff
@@ -7697,6 +7697,137 @@ def test_ld_library_path(self, args):
     self.assertContained('Hello4', out)
     self.assertContained('Ok', out)
 
+  @also_with_wasmfs
+  def test_dlopen_library_path(self):
```

Comment:
How is this different to the existing `test_ld_library_path`?

Reply:
I don't see any important differences, so I think we can remove this one.

Reply:
Oh, the new behavior is that we now correctly handle dependencies between dynamic libraries -- before, if a lib depends on another one, the dlopened lib could be loaded from `LD_LIBRARY_PATH` but not the dependency.

Reply:
Can you extend the existing test maybe?

Reply:
I combined the two tests into one.